crossbeam-epoch/tests/loom.rs

```diff
@@ -0,0 +1,145 @@
+#![cfg(loom)]
+
+use crossbeam_epoch as epoch;
+
+use epoch::*;
+use epoch::{Atomic, Owned};
+use loom::sync::atomic::Ordering::{self, Acquire, Relaxed, Release};
+use loom::sync::Arc;
+use loom::thread::spawn;
+use std::mem::ManuallyDrop;
+use std::ptr;
+
+#[test]
+fn it_works() {
+    loom::model(|| {
+        let collector = Collector::new();
+        let item: Atomic<String> = Atomic::from(Owned::new(String::from("boom")));
+        let item2 = item.clone();
+        let collector2 = collector.clone();
+        let guard = collector.register().pin();
+
+        let jh = loom::thread::spawn(move || {
+            let guard = collector2.register().pin();
+            guard.defer(move || {
+                // this isn't really safe, since other threads may still have pointers to the
+                // value, but in this limited test scenario it's okay, since we know the test won't
+                // access item after all the pins are released.
+                let mut item = unsafe { item2.into_owned() };
+                // mutate it as a second measure to make sure the assert_eq below would fail
+                item.retain(|c| c == 'o');
+                drop(item);
+            });
+        });
+
+        let item = item.load(Ordering::SeqCst, &guard);
+        // we pinned strictly before the call to defer_destroy,
+        // so item cannot have been dropped yet
+        assert_eq!(*unsafe { item.deref() }, "boom");
+        drop(guard);
+
+        jh.join().unwrap();
+
+        drop(collector);
+    })
+}
+
+#[test]
+fn treiber_stack() {
+    // this is mostly a copy-paste from the example
```

Comment:
Maybe we can refactor them to have only one copy of stack-

Reply:
Hmm, I don't know of a way to grab code from an example into another file. We could move the code from the example file into the library, but then it'd no longer be an example. Maybe we could do some hacking with `include!`, but I suspect that would be far uglier. If you have ideas I'm all ears!

Reply:
How about just move the treiber's stack to this file?  I'm fine with removing the examples altogether.  @taiki-e ?

Reply:
I'm also fine with removing the example.